### ISSUE TITLE
Don't load all facilities into session (ever? at least for central server...)

### ISSUE THREAD

**Issue by jamalex** [learningequality/ka-lite#4720]
- The following code, in kalite/facility/middleware.py, loads data for all facilities into the session object:

```Python
# To enable simplified login, also list the id and names of all facilities in the session object.
# This keeps it cached so that the status api call can return this to the client side
# without significantly increasing DB load on every status call.
request.session["facilities"] = [{"id": id, "name": name} for id, name in Facility.objects.values_list("id", "name")]
```

- The session object is persisted, either in the DB or in temp files.

- On the central server, we have 6K+ facilities, and many sessions

- The central server DB is now 7GB+, and we are no longer doing backups because the 500GB backup drive overflowed.

@rtibbles Is that code still required? I can't see any references to `request.session["facilities"]` elsewhere. If it *is* required, we should find a way to disable it on the central server, where it definitely isn't required.

**Comment by rtibbles** [comment]
Used in Base template to populate facility list for login, I believe.

**Comment by jamalex** [comment]
Hmm, ok, would be good to double-check -- couldn't find it when I looked. We don't use that on the central server, at the very least -- or shouldn't.

Do we actually gain any efficiency by storing this in the session, though, in any case? It's stored separately for each visitor, so it's still being computed once per visit anyway, right?

**Comment by benjaoming** [comment]
Why is it in a session? And then loaded upon every request?

Couldn't it just be `setattr(request, 'KALITE_FACILITIES', [{"id": id, "name": name} for id, name in Facility.objects.values_list("id", "name")])

...it could also be done using a template context loader.

But even better: Only doing it on-demand, caching it and invalidating the cache in Facility.save ?

**Comment by rtibbles** [comment]
Fixed in #4912
